tools/bin2c/bin2c.c

```diff
@@ -96,24 +101,28 @@ void generate_transformed_name(const char *path, const char *dir_out)
     }
 
     // Create the array name. If the name begins by a digit, prefix it with a
-    // '_' to form a valid identifier. Also, convert dots and dashes to
-    // underscores.
+    // '_' to form a valid identifier. Also, convert characters which are not
+    // valid C identifier characters underscores.
     //
     //     my-bin-file.bin -> my_bin_file_bin[]
     //     34.my.bin.file.bin -> _34_my_bin_file_bin[]
+    //     my-fancy-file%.bin -> my_fancy_file__bin[]
     {
         const char *prefix = "";
         if (isdigit(path[start]))
             prefix = "_";
 
-        snprintf(out_array_name, sizeof(out_array_name), "%s%s",
-                 prefix, &(path[start]));
-
-        len = strlen(out_array_name);
+        len = snprintf(out_array_name, sizeof(out_array_name), "%s%s",
+               prefix, &(path[start]));
+        if (len < 0 || len >= (int) sizeof(out_array_name))
+        {
+            fprintf(stderr, "Output array name too long\n");
+            exit(1);
+        }
 
-        for (size_t i = 0; i < len; i++)
+        for (int i = 0; i < len; i++)
         {
-            if ((out_array_name[i] == '.') || (out_array_name[i] == '-'))
+            if (!isalnum(out_array_name[i]))
```

Comment:
Yeah, I guess that this makes sense. I was thinking about imitating ``bin2s``, but it doesn't behave in a way that we should copy: https://github.com/devkitPro/general-tools/blob/a980819d1f0db6702b46fcc699911306c2f013b4/bin2s.c#L73-L75